### PR TITLE
Allow specification of timezones in Periodic Jobs

### PR DESCRIPTION
**Pull request by dadgar**
This PR allows specification of time zones in periodic jobs.

Fixes https://github.com/hashicorp/nomad/issues/2276

### CODE REVIEW

File: nomad/structs/structs.go [hunk 1]
```diff
@@ -1558,23 +1572,41 @@ func (p *PeriodicConfig) Validate() error {
...
 
-	return nil
+	return mErr.ErrorOrNil()
+}
+
+func (p *PeriodicConfig) Canonicalize() {
+	// Load the location
+	l, err := time.LoadLocation(p.TimeZone)
+	if err != nil {
+		panic(fmt.Sprintf("Bad location %q: %v", p.TimeZone, err))
```
Comment (briansorahan) [inline nomad/structs/structs.go]: Why panic? This panic should at least be mentioned in a doc comment.

Comment (briansorahan) [inline nomad/structs/structs.go]: Another place where documenting the panic seems relevant https://github.com/hashicorp/nomad/blob/f4cc46948f1893c3b4afc090c76f41abecefeb24/nomad/structs/structs.go#L1164

**Comment by nugend** [comment]
Thanks! The capability looks like it will do just what I was hoping for. Can't comment on implementation details though.

File: nomad/structs/structs.go (hunk 1 above)
Comment (dadgar) [inline nomad/structs/structs.go]: @briansorahan The panic acts as an assertion. If it is hit there is a bug in implementation. The reason it is okay is Validate should check for a valid timezone: https://github.com/hashicorp/nomad/pull/2321/files/f4cc46948f1893c3b4afc090c76f41abecefeb24#diff-396d9428285e4245041a3d0b15546d5dR1581

So the system should never even canonicalize an invalid spec.

Comment (briansorahan) [inline nomad/structs/structs.go]: Are you saying that Validate guards against an invalid timezone when a periodic job is created with the CLI and HTTP?
If so then maybe my concern goes away.
I'm not very familiar with the nomad source code, but I get worried when I see a panic in code that could be part of a long-running process.

**Comment by nugend** [comment]
Shouldn't this log something if it's defaulting to UTC? I don't know exactly how a bad location would get in there, and I think defaulting is sensible, but that's gonna be a heck of a confusing bug if it ever does happen.

**Pull request merged**